CHANGELOG.md

```diff
@@ -1,5 +1,59 @@
-# Changelog
-0.9.0 (2018-05-14)
-==================
+### [Next Release Here]
 
-First beta release of libledger-core for Ledger Vault integration
\ No newline at end of file
+> 2019/??/??
+
+- Change the internal representation of `DynamicObject`. The new representation is optimized to
+  take less memory and to be safer in terms of typing. The internal interfaces were also
```

Comment:
IMHO be better: '... is optimized not only memory wise but also type safety wise '